src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriter.cs

```diff
@@ -823,8 +823,11 @@ protected virtual void FlushBuffer()
                     }
                     else
                     {
-                        // Write text to TextWriter
-                        writer.Write(bufChars, 1, bufPos - 1);
+                        if (bufPos - 1 > 0)
```

Comment:
why wasn't this condition needed before?

Reply:
It is related to https://github.com/dotnet/runtime/issues/34130, its not like not needed before, we were just calling writer.Write(...); or stream.Write(...) even there were nothing to write